tests/2d_tests.cpp

```diff
@@ -0,0 +1,311 @@
+#include <wmtk/TriMesh.h>
+
+#include <stdlib.h>
+#include <catch2/catch.hpp>
+#include <iostream>
+
+using namespace wmtk;
+
+TEST_CASE("load mesh from file and create TriMesh", "[test_mesh_creation]")
+{
+    TriMesh m;
+    std::vector<std::array<size_t, 3>> tris = {{{0, 1, 2}}};
+    m.create_mesh(3, tris);
+    REQUIRE(m.get_tri_connectivity().size() == tris.size());
+    REQUIRE(m.get_vertex_connectivity().size() == 3);
+}
+
+TEST_CASE("test generate tuples with 1 triangle", "[test_tuple_generation]")
+{
+    TriMesh m;
+    std::vector<std::array<size_t, 3>> tris = {{{0, 1, 2}}};
+    m.create_mesh(3, tris);
+    std::vector<TriMesh::TriangleConnectivity> m_tri_connectivity = m.get_tri_connectivity();
+    std::vector<TriMesh::VertexConnectivity> m_vertex_connectivity = m.get_vertex_connectivity();
+    SECTION("test generation from vertics")
+    {
+        auto vertices_tuples = m.generate_tuples_from_vertices();
+        REQUIRE(vertices_tuples.size() == 3);
```

Comment:
You can use `REQUIRE_EQ(x, 3)`

Reply:
Catch2 doesn't have a REQUIRE_EQ afaik. Are you confusing it with Google Test?

Reply:
Yes, I am constantly confused by various testing frameworks :)